app/models/page.rb

```diff
@@ -18,6 +18,16 @@ def path
     is_top_level_path ? "/#{slug}" : "/page/#{slug}"
   end
 
+  def local_html?
+    !Rails.env.production? && File.file?(local_path.to_s)
+  end
+
+  def html
+    return File.read(local_path) if local_html?
```

Comment:
Instead of saving paths to the database, since this is just for local development could we just have reading a local file default to a directory in the project? For example: if the page is named `codeland`, just have the local option look for a file at `<project>/tmp/local/codeland.html` (or a different folder added to `.gitignore`)? That way NOTHING leaks into production!

Reply:
I very much see the value of this! I hadn't thought of a good default and also at some point also considered having an ENV variable that can be set for this purpose.

You just sparked an idea in me for this though 🙏🏼  I'll push some changes soon!